suggestionBox/suggestionBox-war/src/main/java/org/silverpeas/components/suggestionbox/common/SuggestionBoxWebManager.java

```diff
@@ -326,7 +326,7 @@ private SuggestionEntity validateSuggestion(SuggestionBox suggestionBox, Suggest
       }
       ContributionValidation validation =
           new ContributionValidation(newStatus, fromUser, new Date(), validationComment);
-      suggestion.setLastUpdater(fromUser);
+      suggestion.updatedBy(fromUser);
       Suggestion actual = suggestionBox.getSuggestions().validate(suggestion, validation);
       switch (actual.getValidation().getStatus()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Complete cases by adding the missing enum constants or add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Switch statement found in org.silverpeas.components.suggestionbox.common.SuggestionBoxWebManager.validateSuggestion(SuggestionBox, Suggestion, ContributionStatus, String, User) where default case is missing [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=findbugs%3ASF_SWITCH_NO_DEFAULT)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1301)